rust/agama-cli/src/logs.rs

```diff
@@ -45,6 +51,42 @@ pub async fn run(subcommand: LogsCommands) -> anyhow::Result<()> {
     }
 }
 
+// Whatewer passed in dest formed into an absolute path with archive name
+// if dest is none then a default is returned
+// if dest is directory then a default file name for the archive will be appended
+// if dest is path with a file name then it is used as is a name for resulting archive
+fn parse_dest(dest: Option<PathBuf>) -> Result<PathBuf, io::Error> {
+    let default = PathBuf::from(DEFAULT_RESULT);
+    let err = io::Error::new(io::ErrorKind::InvalidInput, "Invalid destination path");
+
+    match dest {
+        None => Ok(default),
```

Comment:
What if `/tmp/agama_logs` is already a directory? What if tomorrow we decide to change the default value? I would run the same checks whether it is the default or a value from the user.

This way, we could drop the `match` and save some nesting:

```rust
let mut buff = dest.unwrap_or(PathBuf::from(DEFAULT_RESULT));
let path = buff.as_path();
```

Reply:
BTW, why not `buff` instead of `buffer`? We are just saving two letters :-)

Reply:
if ```/tmp/agama_logs``` is already directory, then you will end up with ```/tmp/agama_logs.tar.bzip2``` archive.

Reply:
regarding ```buff`` well ... just matter of habit as i have been using ```buff```` and ```dest``` since I was at school, so it is kind of convention for me ;-) It is even used in man pages of such tools like ```cp```,  ```rsync```, e.g. ```rsync``` defines its usage as ```rsync [OPTION...] SRC... [DEST]``` even in some options like ```--copy-dest=DIR```.

However, I don't have problem to extend it. It definitely is not stopper for me  ;-)

Reply:
Not even for me, I was curious especially about the "buff" case :-)

Reply:
About the `/tmp/agama_logs`, my point is that I would run *the same checks* in both cases. And in that case, you could save the `match` statement (too much nesting for my taste).